crowbar_framework/app/models/service_object.rb

```diff
@@ -1418,6 +1440,48 @@ def apply_role(role, inst, in_queue)
     [200, {}]
   end
 
+  def delete_remove_role_from_runlist(nodes_with_roles_to_remove)
+    databag = Chef::DataBag.load "crowbar/bc-template-#{@bc_name}"
+    nodes_with_roles_to_remove.each do |node_name, roles_to_remove|
+      node = NodeObject.find_node_by_name(node_name)
+      next if node.nil?
+
+      # delete #{role_name}_remove role from runlist as it is not needed anymore
+      roles_to_remove.each do |role_to_remove|
+        next unless node.role? role_to_remove
+        @logger.debug("Removing temporary role #{role_to_remove} from #{node.name}")
+        node.delete_from_run_list role_to_remove
+        save_node_state(node, __method__)
+
+        # delete _remove roles from databag
+        databag["deployment"][@bc_name]["elements"]["to_remove"][node_name].delete(role_to_remove)
+        databag.save
+      end
+
+      # delete nodes from databag with no more _remove roles
+      if databag["deployment"][@bc_name]["elements"]["to_remove"][node_name].empty?
+        databag["deployment"][@bc_name]["elements"]["to_remove"].delete(node_name)
+        databag.save
+      end
+    end
+
+    # delete remove intention from databag, as it has been done successfully
+    if databag["deployment"][@bc_name]["elements"]["to_remove"].empty?
+      databag["deployment"][@bc_name]["elements"].delete("to_remove")
```

Comment:
as we are (ab)using the `bc-template-#{@bc_name}` databag I would like to get it back to how it was before. this is the hacky solution that I came up with by @vuntz suggestion [here](https://github.com/crowbar/barclamp-crowbar/pull/1224#discussion_r27341071)

otherwise if there is just **one** point in time, where some role has to be removed, we will have this `to_remove` attribute in the databag under `elements` forever, even though it may not be used anymore.

So this is needed as a cleanup.